ts/interactions/conversationInteractions.ts

```diff
@@ -564,3 +710,89 @@ export async function callRecipient(pubkey: string, canCall: boolean) {
     await CallManager.USER_callRecipient(convo.id);
   }
 }
+
+/**
+ * Updates the interaction state for a conversation. Remember to run clearConversationInteractionState() when the interaction is complete and we don't want to show it in the UI anymore.
+ * @param conversationId id of the converation we want to interact with
+ * @param type the type of conversation interaciton we are doing
+ * @param status the status of that interaction
+ */
+export async function updateConversationInteractionState({
+  conversationId,
+  type,
+  status,
+}: {
+  conversationId: string;
+  type: ConversationInteractionType;
+  status: ConversationInteractionStatus;
+}) {
+  const convo = getConversationController().get(conversationId);
+  if (convo) {
+    convo.set('lastMessageInteractionType', type);
+    convo.set('lastMessageInteractionStatus', status);
+
+    await convo.commit();
+    window.log.debug(
+      `updateConversationInteractionState for ${conversationId} to ${type} ${status}`
+    );
+  }
+}
+
+/**
+ * Clears the interaction state for a conversation. We would use this when we don't need to show anything in the UI once an action is complete.
+ * @param conversationId id of the conversation whose interaction we want to clear
+ */
+export async function clearConversationInteractionState({
+  conversationId,
+}: {
+  conversationId: string;
+}) {
+  const convo = getConversationController().get(conversationId);
+  if (convo) {
+    convo.set('lastMessageInteractionType', undefined);
+    convo.set('lastMessageInteractionStatus', undefined);
+
+    await convo.commit();
+    window.log.debug(`clearConversationInteractionState for ${conversationId}`);
+  }
+}
+
+async function saveConversationInteractionErrorAsMessage({
+  conversationId,
+  interactionType,
+}: {
+  conversationId: string;
+  interactionType: ConversationInteractionType;
+}) {
+  const conversation = getConversationController().get(conversationId);
+  if (!conversation) {
+    return;
+  }
+
+  const interactionStatus = ConversationInteractionStatus.Error;
+
+  await updateConversationInteractionState({
+    conversationId,
+    type: interactionType,
+    status: interactionStatus,
+  });
+
+  // NOTE at this time we don't have visible control messages in communities
+  if (conversation.isPublic()) {
+    return;
+  }
+
+  // Add an error message to the database so we can view it in the message history
+  await conversation?.addSingleIncomingMessage({
+    source: UserUtils.getOurPubKeyStrFromCache(),
+    sent_at: Date.now(),
```

Comment:
`GetNetworkTime.getNowWithnetworkOffset` would be better here

Reply:
Fixed